src/resource/image.rs

```diff
@@ -111,11 +111,12 @@ impl<A: Allocator> Image<A> {
     pub fn new(
         device: Device,
         alloc: &mut A,
-        width: u32,
-        height: u32,
+        extent: vk::Extent3D,
         usage: vk::ImageUsageFlags,
         format: vk::Format,
         samples: vk::SampleCountFlags,
+        mip_levels: u32,
+        layers: u32,
```

Comment:
The image constructor is becoming relatively bloated. If we are going to break API here it's worth considering moving over to a `CreateInfo` struct with possibly an accompanying builder struct.

Reply:
That works yep. Something like ImageCreateInfo?

Reply:
Yeah, exactly